### PR TITLE
Remove location as required property for Resource

### PR DESCRIPTION
**Pull request by vivsriaus**

### CODE REVIEW

File: src/ResourceManagement/Resource/ResourceManagement/Generated/Models/DeploymentPreFlightResource.cs [hunk 1]
```diff
@@ -61,19 +61,5 @@ public DeploymentPreFlightResource()
         {
             this.DependsOn = new LazyList<string>();
         }
-        
-        /// <summary>
-        /// Initializes a new instance of the DeploymentPreFlightResource class
-        /// with required arguments.
-        /// </summary>
-        public DeploymentPreFlightResource(string location)
```
Comment (markcowl) [inline src/ResourceManagement/Resource/ResourceManagement/Generated/Models/DeploymentPreFlightResource.cs]: This is a breaking change., Why not just remove the required check?

File: src/ResourceManagement/Resource/ResourceManagement/Generated/Models/GenericResource.cs [hunk 2]
```diff
@@ -70,19 +70,5 @@ public string ProvisioningState
         public GenericResource()
         {
         }
-        
-        /// <summary>
-        /// Initializes a new instance of the GenericResource class with
-        /// required arguments.
-        /// </summary>
-        public GenericResource(string location)
```
Comment (markcowl) [inline src/ResourceManagement/Resource/ResourceManagement/Generated/Models/GenericResource.cs]: Same comment

File: src/ResourceManagement/Resource/ResourceManagement/Generated/Models/GenericResourceExtended.cs [hunk 3]
```diff
@@ -70,19 +70,5 @@ public string ProvisioningState
         public GenericResourceExtended()
         {
         }
-        
-        /// <summary>
-        /// Initializes a new instance of the GenericResourceExtended class
-        /// with required arguments.
-        /// </summary>
-        public GenericResourceExtended(string location)
```
Comment (markcowl) [inline src/ResourceManagement/Resource/ResourceManagement/Generated/Models/GenericResourceExtended.cs]: same comment

File: src/ResourceManagement/Resource/ResourceManagement/Generated/Models/DeploymentPreFlightResource.cs (hunk 1 above)
Comment (vivsriaus) [inline src/ResourceManagement/Resource/ResourceManagement/Generated/Models/DeploymentPreFlightResource.cs]: @markcowl This is generated code. Are you suggesting I hand-edit these files?

**Comment by vivsriaus** [comment]
@markcowl per our discussion, I've added the customized files to avoid the breaking change.

File: src/ResourceManagement/Resource/ResourceManagement/Generated/Models/DeploymentPreFlightResource.cs (hunk 1 above)
Comment (markcowl) [inline src/ResourceManagement/Resource/ResourceManagement/Generated/Models/DeploymentPreFlightResource.cs]: @vivsriaus No, as we discussed offline, this should be handled through a customization of the partial class.

Comment (vivsriaus) [inline src/ResourceManagement/Resource/ResourceManagement/Generated/Models/DeploymentPreFlightResource.cs]: @markcowl this? https://github.com/Azure/azure-sdk-for-net/commit/c860ffda5d6190163cdd7310088c2cfb050027e3

**Comment by vivsriaus** [comment]
@markcowl can you please take a look at this PR and let me know if it looks good?

**Pull request merged**